### DISCUSSION TITLE
Is it possible to get the path of the URL and use that as the params instead of the slug?

### DISCUSSION THREAD

**Discussion by 684efs3** [remix-run/remix#2341] [Q&A] [answered]
I am fetching data from an API, including the folder structure. Because the folder structure is unknown, I cannot create the folder under routes. Instead I would like to fetch, e.g. "class-a/this-is-interesting" and pass that to loader through params. However, params will not take anything with a slash, /, and will only get "this-is-interesting."

Is there a way to return the full path (including slashes in the middle) in the loader? Thank you!

**Comment by kiliman** [comment, marked as answer]
Create a *splat* route (`$.tsx`). Then you can get the full path in your loader:
```ts
// routes/blog/$.tsx
export loader: LoaderFunction = async ({request, params}) => {
  const slug = params['*']
  ...
}
```
So for request `/blog/test-series/post-1`, the slug would be `test-series/post-1`

**Reply by 684efs3** [reply]
Thank you!!
